packages/x-tree-view/src/useTreeItem2/useTreeItem2.ts

```diff
@@ -79,13 +89,21 @@ export const useTreeItem2 = <
       instance.handleItemKeyDown(event, itemId);
     };
 
+  const createCotentHandleDoubleClick =
+    (otherHandlers: EventHandlers) => (event: React.MouseEvent & MuiCancellableEvent) => {
+      otherHandlers.onDoubleClick?.(event);
+      if (event.defaultMuiPrevented || checkboxRef.current?.contains(event.target as HTMLElement)) {
```

Comment:
Can this be fired while editing?
In which case doesn't stop the editing? :laughing:

Reply:
https://github.com/user-attachments/assets/9cb40536-e25a-4bab-8935-273cc70d48a9

It's to prevent this from happening. But if we [move the double click event to the label slot](https://github.com/mui/mui-x/pull/13388#discussion_r1688045122), we could remove this 🤔

Reply:
I was not questioning the current state, more wondering if we could have some dumb edge cases where people double click on `content` while editing and it calls `onDoubleClick` and end up in a state we don't want.
Something like:

1. User double click while editing
2. It calls `onBlur` on the label input and toggle editing (editing = false)
3. It calls `onDoubleClick` on the content and toggle editing (editing = true)

But I can't reproduce the bug so maybe it can't happen, we can wait for user to find it I think if it exists.

Reply:
Hmm, I don't think they can 🤔 But never say never 😆